src/software/new_geom/line.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <stdexcept>
+
+#include "software/new_geom/polynomial.h"
+
+class Line final : public Polynomial
+{
+   public:
+    /**
+     * Creates a line with zero y-intercept and zero slope
+     */
+    inline explicit Line() {}
```

Comment:
Can we also have a constructor that creates a line given 2 `Points`?

Reply:
This was previously discussed here:
https://github.com/UBC-Thunderbots/Software/pull/942#discussion_r336836685
https://github.com/UBC-Thunderbots/Software/pull/993#discussion_r339379819

An alternative could be to create a method in `util` to handle creating a line based on two points, and also handle the case where the two points have the same `x` value.